### ISSUE TITLE
Cypress incorrectly validates domain of cookies in 3.5.0

### ISSUE THREAD

**Issue by donotello** [cypress-io/cypress#5656]
### Current behavior:

Cookies are failing to set for subdomains during `cy.request`.
E.g. we call `/auth` of `auth.test.server` and it returns `token` cookie with `.test.server`. In this case cookie is not set.

### Desired behavior:

Cookie is set in above example.

### Steps to reproduce: (app code and test code)

Look at [packages/server/lib/request.coffee#setCookiesOnBrowser](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/request.coffee#L449):

```
return if not tough.domainMatch(cookie.domain, parsedUrl.hostname)
```

And at [`tough-cookie` documentation](https://www.npmjs.com/package/tough-cookie#domainmatchstrdomstrcanonicalizetrue):

```
domainMatch(str,domStr[,canonicalize=true])
Answers "does this real domain match the domain in a cookie?". The str is the "current" domain-name and the domStr is the "cookie" domain-name. Matches according to RFC6265 Section 5.1.3, but it helps to think of it as a "suffix match".
```

So it should be:

```
return if not tough.domainMatch(parsedUrl.hostname, cookie.domain)
```

### Versions

`>=3.5.0`

**Comment by tozes** [comment]
+1, as discussed (here)[https://github.com/cypress-io/cypress/issues/5688#issuecomment-558212709], we have exactly the same issue which is stopping us for upgrading from 3.4.1

**Comment by tozes** [comment]
+1, as discussed [here](https://github.com//issues/5688#issuecomment-558212709), we have exactly the same issue which is stopping us for upgrading from 3.4.1

**Comment by fernandopasik** [comment]
@tozes your link was broken
https://github.com/cypress-io/cypress/issues/5688#issuecomment-558212709

**Comment by donotello** [comment]
@jennifer-shehane , @flotwig , @brian-mann,

Shame on my guys, I've missed another incorrect usage of `domainMatch` in cypress here: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/browsers/cdp_automation.ts#L24

Which basically means that this issue is only partially resolved in #5657. Unfortunately looks like we  have incorrect tests for that code as well: https://github.com/cypress-io/cypress/pull/5816.

I'm not sure if I will be able to pick this up soon enough. Could please one of you reopen this one and do the fix?

@tozes FYI

**Comment by flotwig** [comment]
@donotello That's my bad, I noticed that the tests passed without that patch so I left it out. I forgot to go back and double-check it against `tough-cookie`'s documentation, but I believe you're correct. I'll open a PR.
